lib/rexml/element.rb

```diff
@@ -1276,16 +1276,11 @@ def [](name_or_index)
     #   document.root.attribute("x", "a") # => a:x='a:x'
     #
     def attribute( name, namespace=nil )
-      prefix = nil
-      if namespaces.respond_to? :key
-        prefix = namespaces.key(namespace) if namespace
-      else
-        prefix = namespaces.index(namespace) if namespace
-      end
+      prefix = namespaces.key(namespace) if namespace
```

Comment:
```suggestion
      prefix = namespaces[namespace] if namespace
```

may be faster than `namespaces.key(namespace)`.

Reply:
`namespaces.key(namespace)` means `namespaces.invert[namespace]`. Your suggestion changes the code's meaning.

Reply:
Ah, sorry.